js-pkg/server/src/get_binary.js

```diff
@@ -0,0 +1,58 @@
+const fs = require('fs/promises')
+const path = require('path')
+
+function getSuffix(os_type, os_arch) {
+  if (os_type === 'Windows_NT' && os_arch === 'x64') return 'win-x64.exe.gz'
+  if (os_type === 'Linux' && os_arch === 'x64') return 'linux-x64.gz'
+  //we rely on rosetta to provide emulation on aarch64 for now
+  if (os_type === 'Darwin') return 'macos-x64.gz'
+
+  throw new Error(`Unsupported platform: ${os_type} ${os_arch}`)
+}
+
+function binaryUrl(version, os_type, os_arch) {
+  const suffix = getSuffix(os_type, os_arch)
+  const url = `https://github.com/drifting-in-space/y-sweet/releases/download/v${version}/y-sweet-server-${suffix}`
+  return url
+}
+
+async function downloadFile(url, filename_or_folder = '.') {
+  const zlib = require('zlib')
+  const { Readable } = require('stream')
+  const { finished } = require('stream/promises')
+  const res = await fetch(url)
+  if (res.status === 404) {
+    throw new Error('no such binary')
+  }
+  let tmpdir = await fs.mkdtemp('bin')
```

Comment:
surprisingly, `mkdtemp` alone does not put the file in `/tmp`. To do this in a platform-agnostic way, let's do this:

```js
 mkdtemp(join(tmpdir(), 'y-sweet-'));
```

Reply:
since I wasn't using the tmpfile bit anyway ive removed it,

good to know tho